network/transport/v2/conversation.go

```diff
@@ -82,14 +82,14 @@ type conversationManager struct {
 	mutex                  sync.RWMutex
 	conversations          map[string]*conversation
 	validity               time.Duration
-	lastPeerConversationID map[transport.PeerID]conversationID
+	lastPeerConversationID map[string]conversationID
```

Comment:
using `String()` of `Peer` is a bit dangerous, since it was initially intended to provide a loggable/readable representation of `Peer`, rather than a unique hash of the thing. When someone wants to change the format (to further improve readability, add detail, or less detail) it might break. Format also changes when there's an (authnticated) node DID, so a bit odd for a map key. Maybe better to introduce a separate function (e.g. `Key()` or `Hash()`).

Also, `Address` does not seem to be immutable (it is set on `network/transport/grpc/addressbook.go:95`)